DEVELOPMENT.md

```diff
@@ -1,131 +1,77 @@
+# Build WooCommerce Core and all of its dependencies
+pnpm -- turbo run build --filter='woocommerce'
 
-### Typescript Checking
+# Lint the @woocommerce/components package
+pnpm -- turbo run lint --filter='@woocommerce/components'
 
-Typescript is progressively being implemented in this repository, and you might come across some files that are `.ts` or `.tsx`. By default, a VSCode environment will run type checking on such files that are currently open. 
+# Test all of the @woocommerce scoped packages
+pnpm -- turbo run test --filter='@woocommerce/*'
 
-As of now, some parts of the codebase that were imported from the Woocommerce-Admin repository, into the `plugins/woocommerce-admin/client` directory, still fail Typescript checking. This has been scheduled on the team's backlog to be fixed.
+# Build all of the JavaScript packages
+pnpm -- turbo run build --filter='./packages/js/*'
 
-In order to run type checking across the entire repository, you can run this command in your shell, from the root of this repository:
+# Build everything except WooCommerce Core
+pnpm -- turbo run build --filter='!woocommerce' 
 
-```sh
-pnpm tsc -b tsconfig.base.json
+# Lint everything that has changed since the last commit
+pnpm -- turbo run build --filter='[HEAD^1]'
 ```
 
-For better developer experience, the folder `.vscode/tasks.json` has two VSCode tasks to run these commands automatically as well as to parse the output and highlight the errors in the `Problems` tab and in the file explorer pane. The first task runs it once, the second one runs it in the background upon saving of any modified files. This task is also automatically prompted by VSCode to be run upon opening the folder.
-
-
-## Using Xdebug
-
-Please refer to [WP-ENV official README](https://github.com/WordPress/gutenberg/tree/master/packages/env#using-xdebug) section for setting up Xdebug.
-
-## Overriding the Default Configuration
-
-The default configuration comes with PHP 7.4, WooCommerce 5.0, and a few WordPress config values.
-
-You can create `.wp-env.override.json` file and override the default configuration values.
+## Other Commands
 
-You can find more about `.wp-env.override.json` configuration [here](https://github.com/WordPress/gutenberg/tree/master/packages/env#wp-envoverridejson).
+Outside of the commands in [our turbo.json file](turbo.json), each plugin, package, and tool may have unique scripts in their `package.json` files. In these cases, you can execute those commands using `pnpm {script}` and the same `--filter` syntax as Turborepo.
 
-**Example: Overriding PHP version to 8.0**
+### Examples
 
-Create `.wp-env.override.json` in the root directory with the following content.
+Here are some examples of the commands you will make use of.
 
-```json
-{
-	"phpVersion": "8.0"
-}
-```
-
-**Exampe: Adding a locally installed plugin**
-
-Method 1 - Adding to the `plugins` array
+```bash
+# Add a changelog entry for WooCommerce Core
+pnpm changelog add --filter=woocommerce
```

Comment:
It would be quite useful to have this more prominently featured in the PR template and probably also as a separate document/section of some document, because currently, out of the last 10 PRs, 6 contributors haven't added the changelog and having to request and explain this over and over to them seems suboptimal vs having a place in docs we could point them to. What do you think?

Reply:
I'd actually like to lean into a `pre-push` hook that validates the presence of a changelog entry. It's also in the PR checklist. To be honest, if people aren't reading the checklist, I don't know that prominently featuring this specific command would make much of a difference. That's why I'd rather lean into the hook option.